### PR TITLE
Fix external extensions not loading initially

### LINKED ISSUE #7139: Enabled extension not fully loading on startup
With the GKE-sync extension, on starting Lens, the GKE status bar item does not appear. Disable and then re-enable the extension (or a different extension) to make the status bar item appear.

**Screenshots**

https://user-images.githubusercontent.com/40840436/218170213-aca2610c-35ec-42ec-bff5-ba0f8b77008c.mov

**Environment (please complete the following information):**
- Lens Version: 2023.2.101253-alpha

### PR DESCRIPTION
**Pull request by Nokel81**
fixes https://github.com/lensapp/lens/issues/7139

### CODE REVIEW

File: packages/core/src/renderer/extension-discovery/init.injectable.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  instantiate: (di) => ({
+    id: "initialize-extension-discovery",
+    run: async () => {
+      const extensionDiscovery = di.inject(extensionDiscoveryInjectable);
+
+      await extensionDiscovery.init();
+    },
+    runAfter: di.inject(initializeExtensionLoaderInjectable),
+  }),
+  injectionToken: beforeFrameStartsSecondInjectionToken,
```
Comment (jim-docker) [inline packages/core/src/renderer/extension-discovery/init.injectable.ts:21]: How does one know to use `beforeFrameStartsSecondInjectionToken` over `beforeFrameStartsFirstInjectionToken`?

**Review by jim-docker — APPROVED** [review]
tested on mac. ~This might also fix https://github.com/lensapp/lens/issues/7071? Seemed to for me, but I had both our and their pod menu items in the context menu~. Never mind, I can reproduce that problem

File: packages/core/src/renderer/extension-discovery/init.injectable.ts (hunk 1 above)
Comment (Nokel81) [reply packages/core/src/renderer/extension-discovery/init.injectable.ts:21]: If it depends on something that runs in one of the cluster/main frames ifs above in `packages/core/src/renderer/start-frame/start-frame.injectable.ts`

Comment (Nokel81) [reply packages/core/src/renderer/extension-discovery/init.injectable.ts:21]: Hopefully once we remove iframes this can be simplified.

**Comment by Nokel81** [comment]
Will recreate this PR simpler to fix just the issue (but not via Runnables) because the necessary work to fix the tests is huge

**Pull request closed without merging**